src/irmin-pack/inode.ml

```diff
@@ -671,36 +697,43 @@ struct
         if List.length s <> List.length ps then raise (Duplicated_pointers t);
         if s <> ps then raise (Unsorted_pointers t)
       in
-      let hash v = Bin.V.hash (to_bin_v Total v) in
+      let hash v = Bin.V.hash (to_bin_v Truncated v) in
       let rec aux depth t =
         match t with
-        | Concrete.Value l ->
+        | Concrete.Blinded -> None
+        | Concrete.Values l ->
             check_entries t l;
-            Values (StepMap.of_list (List.map Concrete.of_entry l))
+            Some (Values (StepMap.of_list (List.map Concrete.of_entry l)))
         | Concrete.Tree tr ->
             let entries = Array.make Conf.entries None in
             check_pointers t tr.pointers;
             List.iter
               (fun { Concrete.index; pointer; tree } ->
-                let v = aux (depth + 1) tree in
-                let hash = hash v in
-                if not (hash_equal hash pointer) then
-                  raise (Invalid_hash (hash, pointer, t));
-                let t = { hash = lazy pointer; stable = false; v } in
-                entries.(index) <- Some (Ptr.of_target Total t))
+                match aux (depth + 1) tree with
+                | None -> entries.(index) <- Some (Broken pointer)
```

Comment:
This test seems to fail https://github.com/icristescu/irmin/commit/f4eea14eaeefebf4047b6c3a74ecec06088b0522#diff-dbadcd7de0013fa46dca5ed344f5e516f1e5ad6fb3d05e969834a0c866b18ca9R231 I think because we set `Broken` whenever the hash is `Blinded`?

Reply:
yiikes - there is no way to recover the "stable" hash if we only have a partial proof as we need all the values to compute the hash. Will fix this - but we definitively need flattening to have shorter proofs!

Reply:
Should be fixed by 775ab10

Reply:
Actually I have something simpler -- I'll force push shortly.

Reply:
Now in 083b118